src/exception.cr

```diff
@@ -91,3 +91,16 @@ class DivisionByZero < Exception
     super(message)
   end
 end
+
+# Raised when a feature is not implemented yet or when it cannot be implemented
+# on a certain system or platform
+#
+# ```
+# Sysctl.get_int("net.ipv4.forward") # raises NotImplemented ("Sysctl is not implemented on that platform")
+# ````
+#
+class NotImplemented < Exception
+  def initialize(what : String, reason = "is not implemented")
+    super {what, reason}.join(" ")
+  end
+end
```

Comment:
Shall we keep the NotImplemented exception ? (like ruby & python for example), should I move it to a separate PR ? or should I get rid of it ?